src/main/java/io/pinecone/clients/Pinecone.java

```diff
@@ -98,79 +149,242 @@ public IndexModel createServerlessIndex(String indexName, String metric, int dim
+     */
     public IndexModel createPodsIndex(String indexName, Integer dimension, String environment,
                                       String podType, String metric, CreateIndexRequestSpecPodMetadataConfig metadataConfig) {
-        validatePodIndexParams(indexName, dimension, environment, podType, metric, null, null, null);
-
-
         return createPodsIndex(indexName, dimension, environment, podType, metric, null, null, null, metadataConfig,
                 null);
     }
 
-    // Minimal + source collection
+    /**
+     * Overload for creating a new pod index with environment, podType, and metric, and sourceCollection.
+     *
+     * <pre>
+     *     import io.pinecone.clients.Pinecone;
+     *
+     *     Pinecone client = new Pinecone.Builder("your-api-key").build();
+     *
+     *     client.createPodsIndex("my-index", 1536, "us-east4-gcp", "p1.x2", "euclidean", "my-collection");
+     * </pre>
+     *
+     * @param indexName the name of the index to be created.
+     * @param dimension the number of dimensions for the index.
+     * @param environment the cloud environment where you want the index to be hosted.
+     * @param podType the type of pod to use. A string with one of s1, p1, or p2 appended with a "." and one of x1, x2, x4, or x8.
+     * @param metric the metric type for the index. Must be one of "cosine", "euclidean", or "dot_product".
+     * @param sourceCollection the name of the collection to be used as the source for the index.
+     * @return {@link IndexModel} representing the created serverless index.
+     * @throws PineconeException if the API encounters an error during index creation or if any of the arguments are invalid.
+     */
     public IndexModel createPodsIndex(String indexName, Integer dimension, String environment,
                                       String podType, String metric, String sourceCollection) {
-        validatePodIndexParams(indexName, dimension, environment, podType, metric, null, null, null);
-
         return createPodsIndex(indexName, dimension, environment, podType, metric, null, null, null, null,
                 sourceCollection);
     }
 
-    // Minimal + pods
+    /**
+     * Overload for creating a new pod index with environment, podType, and pods.
+     *
+     * <pre>
+     *     import io.pinecone.clients.Pinecone;
+     *
+     *     Pinecone client = new Pinecone.Builder("your-api-key").build();
+     *
+     *     client.createPodsIndex("my-index", 1536, "us-east4-gcp", "p1.x2", "euclidean", 6);
+     * </pre>
+     *
+     * @param indexName the name of the index to be created.
+     * @param dimension the number of dimensions for the index.
+     * @param environment the cloud environment where you want the index to be hosted.
+     * @param podType the type of pod to use. A string with one of s1, p1, or p2 appended with a "." and one of x1, x2, x4, or x8.
+     * @param pods the number of pods to be used in the index.
```

Comment:
I'd add that the # of `pods` must be equal to `replicas*shards`, and that passing `pods` is mutually exclusive to passing `shards` and `replicas` unless you're in the main method, which accepts all possible params

Reply:
It's called out explicitly in the function which accepts all params, and you can't pass shards or replicas in this function so probably not worth calling out here.